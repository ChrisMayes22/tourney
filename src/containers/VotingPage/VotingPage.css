footer{
    grid-area: footer;
    text-align: center;
}

.modalBackground {
    position: fixed;
    height: 100vh;
    width: 100vw;
    z-index: 99;
    background-color: rgba(0, 0, 0, 0.3);
}

.modalFlexContainer{
    margin: 0 auto;
    display: flex;
    justify-content: space-around;
    height: 25vh;
    width: 70vw;
}

.modalSizing{
    top: 14vh;
    left: 10vw;
    height: 60vh;
    width: 80vw;
}

.modalImage{
    border-radius: 8px;
    background-color: rgb(44, 9, 9);
    max-height: 20vh;
    max-width: 30vw;
    box-shadow: 2px 2px 10px rgb(0, 0, 0);
}
