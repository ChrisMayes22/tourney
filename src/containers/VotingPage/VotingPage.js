import React, { Component } from 'react';
import classes from './VotingPage.css';
import SubmitButton from '../../components/SubmitButton/SubmitButton';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { submitCharacterRatings, rateCharacter, chooseWinner, resetForFinals, removeLowestScore } from '../../constants/actions';
import VotingLayout from '../../components/VotingLayout/VotingLayout';
import PrelimVotingContent from '../../components/PrelimVotingContent/PrelimVotingContent';
import FinalsVotingContent from '../../components/FinalsVotingContent/FinalsVotingContent';
import Modal from '../../components/Modal/Modal';
import * as urls from '../../constants/urls';

export class VotingPage extends Component {

    state = {
        modal: true,
    }

    toggleModalHandler = () =>{
        const displayToggle = !this.state.modal;
        this.setState({
            modal: displayToggle
        })
    }

    startVotingHandler = () => {
        this.toggleModalHandler();
        this.props.onClearModal();
    }

    submitWinnerHandler = (player) => {
        this.props.onChooseWinner(player);
        this.toggleModalHandler();
    }

    playersArray= this.props.characters.filter(char => !char.hadTurn)

    render () {

        return(
            <React.Fragment>
                {!(this.state.modal * this.props.finals) ? null :
                    <Modal sizing={classes.modalSizing}>
                        <h1>Only these few remain!</h1>
                        <h5>From now on, matches will be single elimination.</h5>
                        <div className={classes.modalFlexContainer}>
                            {this.props.characters.filter(el => !el.isEliminated.check).map(char => {
                                return <img 
                                    src={char.imageUrl} 
                                    key={`charImage:${char.id}`} 
                                    alt={'competitor'} 
                                    className={classes.modalImage}/>
                            })}
                        </div>
                        <h5>Are you ready to begin?</h5>
                        <br/>
                        <SubmitButton 
                            children={'Yes!'}
                            clicked={() => this.startVotingHandler()}/>      
                    </Modal>}

                <VotingLayout
                    finals={this.props.finals ? true : false}
                    players={this.props.players}>
                    {this.props.finals ?
                        <FinalsVotingContent
                            characters={this.props.characters}
                            players={this.props.players}
                            clicked={this.submitWinnerHandler}/> :
                        <PrelimVotingContent
                            characters={this.props.characters}
                            players={this.props.players}
                            clicked={this.props.onRate}/>}
                </VotingLayout>

                {this.props.finals ? null :
                    <footer>
                        <Link 
                            onClick={() => this.props.onSubmitRatings(this.playersArray)}
                            to={urls.UPLOAD_PAGE}>
                                <SubmitButton children={"Submit"}/>
                        </Link>
                    </footer>}
            </React.Fragment>
        );
    }
}


const mapStateToProps = state => {
    return {
        characters: state.characters,
        players: state.players,
        finals: state.finals
    }
};

const mapDispatchToProps = dispatch => {
    return {
        onChooseWinner: (character) => dispatch(chooseWinner(character)),
        onClearModal: () => dispatch(resetForFinals()),
        onEliminate: () => dispatch(removeLowestScore()),
        onSubmitRatings: (playersArray) => dispatch(submitCharacterRatings(playersArray)),
        onRate: (points, rowId, character) => dispatch(rateCharacter(points, rowId, character))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(VotingPage);