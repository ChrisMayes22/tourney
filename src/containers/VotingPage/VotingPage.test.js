import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter(), disableLifecycleMethods: true });

import React from 'react';
import { shallow } from 'enzyme';

import { VotingPage } from './VotingPage';

import configureStore from 'redux-mock-store';
const middleware = [];
const mockStore = configureStore(middleware);

describe('When unconnected component first renders', function(){
    test('VotingPage renders as expected', () => {
      const props = {characters:[], players:4}
      const wrapper = shallow(<VotingPage {...props}/>)
    
      expect(wrapper).toMatchSnapshot();
    })
    // describe('Characters map renders properly', function(){
    //   test('when characters map renders, only characters for which hadTurn:true render', function(){
    //       let props = {characters: [{hadTurn: false, id: 'foo$id', imageUrl: 'foo$image'},
    //                                 {hadTurn: false, id: 'foo$id1', imageUrl: 'foo$image'},
    //                                 {hadTurn: false, id: 'foo$id', imageUrl: 'foo$image'},
    //                                 {hadTurn: false, id: 'foo$id1', imageUrl: 'foo$image'},
    //                                 {hadTurn: true, id: 'foo$id', imageUrl: 'foo$image'},
    //                                 {hadTurn: true, id: 'foo$id1', imageUrl: 'foo$image'},
    //                                 {hadTurn: true, id: 'foo$id', imageUrl: 'foo$image'},
    //                                 {hadTurn: true, id: 'foo$id1', imageUrl: 'foo$image'}], 
    //                     players: 4}
    //       const wrapper = shallow(<VotingPage {...props}/>);

    //       expect(wrapper.find('[character]')).toHaveLength(4);
    //   })
    // })
})

//TODO: Combine testing for what used to be FinalsPage w/ new VotingPage


// import { configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// configure({ adapter: new Adapter(), disableLifecycleMethods: true });

// import React from 'react';
// import { shallow } from 'enzyme';

// import { FinalsPage } from './FinalsPage';

// let props = null;
// let wrapper = null;
// let component = null;

// describe('When unconnected FinalsPage first renders', function(){
//     beforeEach(function(){
//         props = {characters: [{
//             imageUrl: `foo$Image1`,
//             roundRatings: {
//                 rowOne: {rowId: `foo$id1-row-1`, points: 5},
//                 rowTwo: {rowId: `foo$id1-row-2`, points: 5},
//                 rowThree: {rowId: `foo$id1-row-3`, points: 5},
//                 rowFour: {rowId: `foo$id1-row-4`, points: 5}
//             },
//             points: 20,
//             hadTurn: false,
//             isEliminated: {check: false, whenEliminated: null},
//             id: 'foo$id1'
//           },
//           {
//             imageUrl: `foo$Image2`,
//             roundRatings: {
//                 rowOne: {rowId: `foo$id2-row-1`, points: 1},
//                 rowTwo: {rowId: `foo$id2-row-2`, points: 1},
//                 rowThree: {rowId: `foo$id2-row-3`, points: 1},
//                 rowFour: {rowId: `foo$id2-row-4`, points: 1}
//             },
//             points: 4,
//             hadTurn: false,
//             isEliminated: {check: false, whenEliminated: null},
//             id: 'foo$id2'
//           },
//           {
//             imageUrl: `foo$Image3`,
//             roundRatings: {
//                 rowOne: {rowId: `foo$id3-row-1`, points: 2},
//                 rowTwo: {rowId: `foo$id3-row-2`, points: 2},
//                 rowThree: {rowId: `foo$id3-row-3`, points: 2},
//                 rowFour: {rowId: `foo$id3-row-4`, points: 2}
//             },
//             points: 8,
//             hadTurn: false,
//             isEliminated: {check: false, whenEliminated: null},
//             id: 'foo$id3'
//           }],
//           players: 3
//         }
        
//         wrapper = shallow(<FinalsPage {...props}/>)
//         component = wrapper.instance();
//     })
//     test('FinalsPage renders as expected', function(){
//         expect(wrapper).toMatchSnapshot();
//     })
//     describe('When state initializes', function(){
//         test('state.modal is true', function(){
//             expect(component.state.modal).toBe(true);
//         })
//     })
//     describe('When character map in modal is initialized', function(){
//         test('The number of characters rendered === number of non-eliminated characters', function(){
//             expect(wrapper.find('[alt="competitor"]')).toHaveLength(3);
//         });
//         test('eliminated characters should not be rendered in modal', function(){
//             props = {characters: [{
//                     imageUrl: `foo$Image1`,
//                     roundRatings: {
//                         rowOne: {rowId: `foo$id1-row-1`, points: 5},
//                         rowTwo: {rowId: `foo$id1-row-2`, points: 5},
//                         rowThree: {rowId: `foo$id1-row-3`, points: 5},
//                         rowFour: {rowId: `foo$id1-row-4`, points: 5}
//                     },
//                     points: 20,
//                     hadTurn: false,
//                     isEliminated: {check: false, whenEliminated: null},
//                     id: 'foo$id1'
//                 },
//                 {
//                     imageUrl: `foo$Image2`,
//                     roundRatings: {
//                         rowOne: {rowId: `foo$id2-row-1`, points: 1},
//                         rowTwo: {rowId: `foo$id2-row-2`, points: 1},
//                         rowThree: {rowId: `foo$id2-row-3`, points: 1},
//                         rowFour: {rowId: `foo$id2-row-4`, points: 1}
//                     },
//                     points: 4,
//                     hadTurn: false,
//                     isEliminated: {check: true, whenEliminated: 0},
//                     id: 'foo$id2'
//                 }],
//                 players: 3
//             };
//             wrapper = shallow(<FinalsPage {...props}/>)
//             component = wrapper.instance();

//             expect(wrapper.exists('[src="foo$Image2"]')).toBe(false);
//             expect(wrapper.exists('[src="foo$Image1"]')).toBe(true);
//             expect(wrapper.find('[alt="competitor"]')).toHaveLength(1);
//         })
//     })
//     describe('When the .grid container is rendered', function(){
//         test('The first character rendered is the first non-eliminated character in props.characters', function(){
//             expect(wrapper.find('#first-char').prop('character')).toEqual(props.characters[0]);
//         })
//         test('The second character rendered is the second non-eliminated character in props.characters', function(){
//             expect(wrapper.find('#second-char').prop('character')).toEqual(props.characters[1]);
//         })
//     })
// })

// describe('When unconnected component methods are invoked', function(){
//     beforeEach(function(){
//         props = {characters: [{
//             imageUrl: `foo$Image`,
//             roundRatings: {
//                 rowOne: {rowId: `foo$id1-row-1`, points: 5},
//                 rowTwo: {rowId: `foo$id1-row-2`, points: 5},
//                 rowThree: {rowId: `foo$id1-row-3`, points: 5},
//                 rowFour: {rowId: `foo$id1-row-4`, points: 5}
//             },
//             points: 20,
//             hadTurn: false,
//             isEliminated: {check: false, whenEliminated: null},
//             id: 'foo$id1'
//           },
//           {
//             imageUrl: `foo$Image`,
//             roundRatings: {
//                 rowOne: {rowId: `foo$id2-row-1`, points: 1},
//                 rowTwo: {rowId: `foo$id2-row-2`, points: 1},
//                 rowThree: {rowId: `foo$id2-row-3`, points: 1},
//                 rowFour: {rowId: `foo$id2-row-4`, points: 1}
//             },
//             points: 4,
//             hadTurn: false,
//             isEliminated: {check: false, whenEliminated: null},
//             id: 'foo$id2'
//           },
//           {
//             imageUrl: `foo$Image`,
//             roundRatings: {
//                 rowOne: {rowId: `foo$id3-row-1`, points: 2},
//                 rowTwo: {rowId: `foo$id3-row-2`, points: 2},
//                 rowThree: {rowId: `foo$id3-row-3`, points: 2},
//                 rowFour: {rowId: `foo$id3-row-4`, points: 2}
//             },
//             points: 8,
//             hadTurn: false,
//             isEliminated: {check: false, whenEliminated: null},
//             id: 'foo$id3'
//           }],
//           players: 3
//         }
    
//         wrapper = shallow(<FinalsPage {...props}/>)
//         component = wrapper.instance();
//     })
    
//     test('Given state.modal === true, when toggleModalHandler is invoked, then state.modal === false', function(){
//         expect(component.state.modal).toBe(true);
//         component.toggleModalHandler();
//         expect(component.state.modal).toBe(false);
//     })
//     test('Given state.modal === false, when toggleModalHandler is invoked, then state.modal === true', function(){
//         wrapper.setState({modal: false});
//         expect(component.state.modal).toBe(false);
//         component.toggleModalHandler();
//         expect(component.state.modal).toBe(true);
//     })
// })