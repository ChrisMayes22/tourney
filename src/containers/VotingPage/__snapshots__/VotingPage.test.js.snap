// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When unconnected component first renders VotingPage renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VotingPage
    characters={Array []}
    players={4}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        null,
        <votingLayout
          finals={false}
          players={4}
        >
          <prelimVotingContent
            characters={Array []}
            players={4}
          />
        </votingLayout>,
        <footer>
          <Link
            onClick={[Function]}
            replace={false}
            to="/upload"
          >
            <submitButton>
              Submit
            </submitButton>
          </Link>
        </footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <prelimVotingContent
            characters={Array []}
            players={4}
          />,
          "finals": false,
          "players": 4,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "characters": Array [],
            "clicked": undefined,
            "players": 4,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            onClick={[Function]}
            replace={false}
            to="/upload"
          >
            <submitButton>
              Submit
            </submitButton>
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <submitButton>
              Submit
            </submitButton>,
            "onClick": [Function],
            "replace": false,
            "to": "/upload",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
          "type": [Function],
        },
        "type": "footer",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          null,
          <votingLayout
            finals={false}
            players={4}
          >
            <prelimVotingContent
              characters={Array []}
              players={4}
            />
          </votingLayout>,
          <footer>
            <Link
              onClick={[Function]}
              replace={false}
              to="/upload"
            >
              <submitButton>
                Submit
              </submitButton>
            </Link>
          </footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <prelimVotingContent
              characters={Array []}
              players={4}
            />,
            "finals": false,
            "players": 4,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "characters": Array [],
              "clicked": undefined,
              "players": 4,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              onClick={[Function]}
              replace={false}
              to="/upload"
            >
              <submitButton>
                Submit
              </submitButton>
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <submitButton>
                Submit
              </submitButton>,
              "onClick": [Function],
              "replace": false,
              "to": "/upload",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": [Function],
            },
            "type": [Function],
          },
          "type": "footer",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
